60_network_learning.Rmd

```diff
@@ -0,0 +1,654 @@
+
+6.  **Transformation into similarity / edge weight:** Finally, the dissimilarities are transformed into similarities by $s_{ij} = 1 - d_{ij}$, which are used as edge weights. Thus, the similarity matrix is equal to the adjacency matrix in a weighted network.
+
+In the following, we show an application for each of the three types of association: correlation, proportionality, and partial correlation. 
+
+We demonstrate the workflow using the the **PeerJ data set** [@potbhare2022skin]. It contains skin microbial profiles of 58 subjects.
+
+```{r, message=FALSE, warning=FALSE}
+library(mia)
+```
+
+```{r load_data}
+data("peerj13075", package = "mia")
+tse <- peerj13075
+dim(tse)
+```
+
+```{r, echo=FALSE}
+# The networks shown in this chapter take several minutes to generate. 
+# Therefore, the network objects have been stored in a folder and are loaded 
+# here to reduce the time needed to knit the book.
+load("general/network_data/networks.RData")
+```
+
+### Install packages
+
+Three packages used in this section are available on GitHub only: `SpiecEasi`, `SPRING`, and `NetCoMi`. We recommend that you install these packages before proceeding.
+
+```{r install_packages}
+if(!require(SpiecEasi)){
+  devtools::install_github("zdk123/SpiecEasi")
+}
+
+if(!require(SPRING)){
+  devtools::install_github("GraceYoon/SPRING")
+}
+
+if(!require(NetCoMi)){
+  devtools::install_github("stefpeschel/NetCoMi", force = TRUE,
+                           dependencies = c("Depends", "Imports", "LinkingTo"),
+                           repos = c("https://cloud.r-project.org/",
+                                     BiocManager::repositories()))
+}
+```
+
+
+### Data preparation
+
+Before we apply the network learning methods, we perform some data preparation steps similar to that in section \@ref(differential-abundance):
+
+-   Aggregate to genus level (would not be necessary here since the data is already at genus level)
+-   Prevalence filtering (keep genera with prevalence \> 20%)
+-   Add assay with clr-transformed abundances
+
+```{r preprocessing}
+# Agglomerate by genus and subset by prevalence
+tse <- subsetByPrevalentFeatures(tse,
+                                 rank = "genus",
+                                 prevalence = 0.2,
+                                 detection = 0)
+
+# Add clr-transformed abundances
+tse <- transformAssay(tse, method="clr", pseudocount=1)
+
+dim(tse)
+```
+
+Since we want to use the phylum names for node colors later in the network plot, we assign low abundant phyla to a new class "Other" to reduce the number of colors needed in the plot.
+
+```{r}
+# Assign low abundant phyla to a "Other" class
+phyla <- rowData(tse)$phylum
+low_abundant <- names(table(phyla)[table(phyla) <= 3])
+phyla[phyla %in% low_abundant] <- "Other"
+rowData(tse)$phylum <- phyla
+
+# Take a look at the table
+table(rowData(tse)$phylum)
+```
+
+
+### SparCC correlation {#sparcc-correlation}
```

Comment:
Because nowadays SPIEC-EASI and other methods are usually replacing SparCC I am wondering if SparCC and some other methods are really needed for the main text. I would separate the material in two categories: 1) recommended practice & 2) educational background. If SparCC is not recommended for most real scenarios (?) then could we think of creating a supplementary document that can be located in OMA and cited in the main text, such document could have examples of methods that we would not recommend for most of the standard use cases but that may still be useful as educational background? 

But we may have different information & thoughts, open to discuss the choices.